apache/cloudstack
# Add, Delete Storage Pool commands should be able execute on a host in maintenance

### Description

This PR...

Fixes #9295 

When a host is in maintenance, CreateStoragePoolCommand and DeleteStoragePoolCommand are not allowed to execute by the AgentAttache.
This will cause a new storage pool to not be present on the host even after it comes out of maintenance, as the cloudstack agent is not restarted when cancel maintain is called (#3239).
This also causes a deleted storage pool to never be removed from a host in maintenance.
















### Types of changes


### Feature/Enhancement Scale or Bug Severity

#### Feature/Enhancement Scale


#### Bug Severity



### Screenshots (if appropriate):


### How Has This Been Tested?

1. Put a host in maintenance.
2. Add a new storage pool and verify that it is mounted on the host
3. Delete a storage pool and verify that it has been unmounted from the host
4. Take the host out of maintenance and verify again.




#### How did you try to break this feature and the system with this change?
